packages/babel-types/src/definitions/experimental.js

```diff
@@ -265,3 +265,37 @@ defineType("BigIntLiteral", {
   },
   aliases: ["Expression", "Pureish", "Literal", "Immutable"],
 });
+
+defineType("SliceExpression", {
+  visitor: ["object", "lower", "upper", "step"],
+  aliases: ["Expression", "LVal"],
+  fields: {
+    object: {
+      validate: assertNodeType("Expression"),
+    },
+    lower: {
+      validate: assertNodeType(
```

Comment:
consider use a custom validator function which aligns with `parser.checkSliceArgument`.

Reply:
I realize that `@babel/parser` and `@babel/types` does not depends on each other, which means it is currently not feasible to share routines between them. I would leave it as-is.